src/network/sniffer.rs

```diff
@@ -148,7 +153,11 @@ impl Sniffer {
             direction,
         })
     }
-    fn handle_v4(ip_packet: Ipv4Packet, network_interface: &NetworkInterface) -> Option<Segment> {
+    fn handle_v4(
+        ip_packet: Ipv4Packet,
+        network_interface: &NetworkInterface,
+        show_dns: bool,
```

Comment:
Hmm... can't we use `self.dns_shown` inside this function instead of passing `show_dns` to it?

Reply:
@imsnif hmm, but `handle_v4` function hasn't `self` parameter, no?

Reply:
You're right. I'm not sure why we implemented it like that *shrugs* nevermind. :)

Reply:
@imsnif ok, so, we keep at as it is, or should i try to modify it and passing `self` parameter ?